### DISCUSSION TITLE
Getting NO DATA error for probe failure.

### DISCUSSION THREAD

**Discussion by Gopinath-31893** [prometheus/prometheus#16872] [Q&A]
Hi,

We have our Prometheus and Blackbox exporter running in an AKS cluster.

We are monitoring machines via blackbox exporter through ICMP probe. 

The targets are pinged from the blackbox exporter and based on the response, it marks as success or failure.

1. If the server is available and responding to ping it shows the result values are probe_success == 1 (UP)
2. If the server is down/unresponsive to ping request. it considered as DOWN and show the result value as probe_success == 0 (DOWN)

Probe_success value denoted the UP and DOWN of the server status (1 denotes the server is UP, 0 denotes the server is DOWN)

But in our case, what happened is. The actual server were down for more than 4 hours. During this time windows instead of showing the probe_success value as **zero**. NO DATA is occurred. 

If we search for the probe_success query in Prometheus it's not showing the probe_success = 0 for the down time. For the entire down time there is no data captured.

After the servers are started, its directly showing the value to 1 (ONE) with probe_success value.
Due to this **NO DATA** occurrence we missed the Alerts. How can get this sorted out.

We are getting this NO DATA occurrence on multiple scenarios. But not this much long period. This would impact our monitoring system. Looking for an suggestion on this.

**Blackbox Exporetr Details:**

We have configured through helm chart

chart Version : prometheus-blackbox-exporter-8. 6.1
Blackbox version: v0.24.0

**Blackbox Config File:**

apiVersion: v1
data:
blackbox.yaml: |
modules:
icmp:
preferred_ip_protocol: ip4
prober: icmp
timeout: 90s

**Prometheus Config file:**

  job_name: 'job1'
  scrape_interval: 5m
  scrape_timeout: 1m
  static_configs:
    targets: ['server1.com']
    metric_path: /probe
    params:
    module: [icmp]

and we have done relabel_config and pointed to blackbox URL for collecting metrics

Please let me know if any additional details are required here.

Thanks,
Gopinath P

**Comment by Gopinath-31893** [comment]
Hi, is there any update on this?? 

This happened in our production environment and due to this behaviour. The server were down for more than 4 hours and it also impacted.

**Comment by bboreham** [comment]
Hello from the bug-scrub.

This looks like a question on usage, for which we suggest resources at https://prometheus.io/community/.
